Automattic/jetpack
# JS Components: Convert SplitButton component to TypeScript

#### Changes proposed in this Pull Request:

* This PR converts `SplitButton` component to TypeScript and cleans up the propTypes.
* It also converts the `SplitButton` stories to TS

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `SplitButton` component work as expected

<img width="872" alt="Screenshot 2022-05-06 at 5 02 37 PM" src="https://user-images.githubusercontent.com/18226415/167125737-5868f297-ca08-4e37-896f-cc3c2a1aac18.png">
